docs/i18n.md

```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
+fancy chained API but we've stuck to a more traditional approach.
+
+Before we get into how to make use of these functions let's take a look at
+how the Jed instance is exposed to our components.
+
+### The Translation Provider
+
+The translation provider is used to pass down a Jed instance via context to
+components lower down in the component hierarchy. This part is already done
+for you in the addons-frontend code-base. So you should only need to worry
+about wrapping your components as detailed in the next section.
+
+## The translate component wrapper.
+
+The translate Higher Order Component is a helper that wraps any component
+and takes the `Jed` `i18n` instance from context and makes it  available in
+the wrapped component's props.
+
+Here's an example of a very basic component setup for translation:
+
+
+```javascript
+import React, { PropTypes } from 'react';
+import translate from 'core/i18n/translate';
+
+
+export class MyTranslatedComponent extends React.Component {
+  static propTypes = {
+    i18n: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { i18n } = this.props;
+    return (
+      <div>
+        <p>{i18n.gettext('Something translated')}</p>
+      </div>
+    );
+  }
+}
+
+export default translate()(MyTranslatedComponent);
+```
+
+That's pretty much all there is to it.
+
+## Using the Jed API
+
+Once you have `i18n` available to your component you can then use
+any of the Jed methods exposed on the `i18n` object.
+
+```
+gettext = function ( key )
+dgettext = function ( domain, key )
+dcgettext = function ( domain, key, category )
+ngettext = function ( singular_key, plural_key, value )
+dngettext = function ( domain, singular_ley, plural_key, value )
+dcngettext = function ( domain, singular_key, plural_key, value, category )
+pgettext = function ( context, key )
+dpgettext = function ( domain, context, key )
+npgettext = function ( context, singular_key, plural_key, value )
+dnpgettext = function ( domain, context, singular_key, plural_key, value )
+dcnpgettext = function ( domain, context, singular_key, plural_key, value, category )
+sprintf = function ( string, substitutions)
+```
+
+### Using sprintf
+
+As you can see a sprintf function is also available. You can use this to
+provide substitutions in gettext wrapped strings.
+
+There are two flavours to this, numbered placeholders or named ones.
+
+Here's the numbered approach:
+
+```javascript
+i18n.sprintf(i18n.gettext('I like your %1$s %2$s.'), 'red', 'shirt'));
+```
+
+and here's the named arg approach:
+
+```javascript
+i18n.sprintf(i18n.gettext('I like your %(colour)s %(garment)s.'), { colour: 'red', garment: 'shirt' }));
+```
+
+Both of these approaches allow for translators to re-order the substitution
+vars.
+
+### Guidance on HTML in translations
+
+Generally we're looking to avoid having HTML in the middle of translation
+strings as much as possible.
+
+If you need HTML it's better to use substitutions to add the HTML than
```

Comment:
Could you include an example of this? It's not how we did it in Django if I recall correctly, and while it sounds safer it also sounds a lot more complicated in my head.